encoders/nlp/TransformerTorchEncoder/tests/test_transformertorchencoder.py

```diff
@@ -224,3 +226,24 @@ def test_no_cls_token(test_metas, params):
     test_data = np.array(['it is a good day!', 'the dog sits on the floor.'])
     with pytest.raises(ValueError):
         _ = encoder.encode(test_data)
+
+
+@pytest.mark.parametrize('pooling_strategy', ['cls', 'mean', 'max'])
+def test_encoding_results_api(test_metas, pooling_strategy):
+    encoder = TransformerTorchEncoder(
+        metas=test_metas,
+        pretrained_model_name_or_path='sentence-transformers/distilbert-base-nli-stsb-mean-tokens',
+        pooling_strategy=pooling_strategy,
+        api_token='fake_token',
+    )
+
+    api_mock = MagicMock(return_value=np.random.rand(2, 9, 768).tolist())
+    encoder._api_call = api_mock
```

Comment:
the _api_call return `json` does not return a `dict`?

Reply:
Wouldn't it be better to mock the `external request` call better than the complete `_api_call` so that that is the code tested. Also we can assert the parameters sent to the API. To make sure what we send to HuggingFace?

Reply:
done